crates/rpc/rpc/src/eth/api/block.rs

```diff
@@ -15,6 +15,44 @@ use reth_rpc_types::{Index, RichBlock, TransactionReceipt};
 
 use reth_rpc_types_compat::block::{from_block, uncle_block_from_header};
 use reth_transaction_pool::TransactionPool;
+
+#[cfg(feature = "optimism")]
+use reth_primitives::U256;
+#[cfg(feature = "optimism")]
+use revm::L1BlockInfo;
+#[cfg(feature = "optimism")]
+use std::rc::Rc;
+
+/// Optimism Transaction Metadata
+///
+/// Includes the L1 fee and data gas for the tx along with the L1
+/// block info. In order to pass the [OptimismTxMeta] into the
+/// async colored `build_transaction_receipt_with_block_receipts`
+/// function, a reference counter for the L1 block info is
+/// used so the L1 block info can be shared between receipts.
+#[cfg(feature = "optimism")]
+#[derive(Debug, Default, Clone)]
+pub(crate) struct OptimismTxMeta {
+    /// The L1 block info.
+    pub(crate) l1_block_info: Option<Rc<L1BlockInfo>>,
+    /// The L1 fee for the block.
+    pub(crate) l1_fee: Option<U256>,
+    /// The L1 data gas for the block.
+    pub(crate) l1_data_gas: Option<U256>,
+}
+
+#[cfg(feature = "optimism")]
+impl OptimismTxMeta {
+    /// Creates a new [OptimismTxMeta].
+    pub(crate) fn new(
+        l1_block_info: Option<Rc<L1BlockInfo>>,
+        l1_fee: Option<U256>,
+        l1_data_gas: Option<U256>,
+    ) -> Self {
+        Self { l1_block_info, l1_fee, l1_data_gas }
+    }
+}
```

Comment:
this file should only contain the EthApi impl.

please move to a separate file

Reply:
Moved to an `optimism.rs` mod in the same directory.

Addressed in https://github.com/anton-rs/op-reth/pull/155